File: Recycler_view/app/src/main/java/com/example/recycler_view/MainActivity.java
package com.example.recycler_view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.Adapter;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
RecyclerView listofproducts;

ArrayList<list_products>list;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        createlist();
        listofproducts=findViewById(R.id.recycler1);
        listofproducts.setAdapter(new product(list));
        RecyclerView.LayoutManager layoutManager=new LinearLayoutManager(this,RecyclerView.VERTICAL,false);
        listofproducts.setLayoutManager(layoutManager);


    }
    public void createlist(){
        list=new ArrayList<list_products>();
        for (int i=1;i<150;i++){
            list.add(new list_products("laptop num:"+i,i*1000+"",R.drawable.ic_launcher_foreground));

        }



    }
}

File: app/src/main/java/com/example/recycler_onclicklistner/MainActivity.kt
package com.example.recycler_onclicklistner

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    lateinit var list:MutableList<setting_item>
    lateinit var products:RecyclerView
    lateinit var adapter:products
    lateinit var layoutManager: RecyclerView.LayoutManager


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initrecycler()
    }

    private fun initrecycler() {
        createlist()
        products=findViewById(R.id.recycler1)
        adapter= products(list)
        products.adapter=adapter
        layoutManager=LinearLayoutManager(this  )
        products.layoutManager=layoutManager
    }

    private fun createlist() {
        list= mutableListOf()
        for (i in 0 until 100){
            list.add(setting_item("hawawi",R.drawable.ic_baseline_1x_mobiledata_24))


        }
    }
}